---
description: '<Description from rule metadata here>'
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> ðŸ›‘ This file is source code, not the primary documentation location! ðŸ›‘
>
> See **https://typescript-eslint.io/rules/RULE_NAME_REPLACEME** for documentation.

This rule reports when a type assertion is used to indicate that an array access may produce undefined due to the index being out of bounds.

## Examples

To fill out: tell us more about this rule.

<Tabs>
<TabItem value="âŒ Incorrect">

```ts
// To fill out: incorrect code
```

</TabItem>
<TabItem value="âœ… Correct">

```ts
// To fill out: correct code
```

</TabItem>
</Tabs>

## When Not To Use It

To fill out: why wouldn't you want to use this rule?
For example if this rule requires a feature released in a certain TS version.
